compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -704,7 +704,13 @@ pub enum TerminatorKind<'tcx> {
     /// The `replace` flag indicates whether this terminator was created as part of an assignment.
     /// This should only be used for diagnostic purposes, and does not have any operational
     /// meaning.
-    Drop { place: Place<'tcx>, target: BasicBlock, unwind: UnwindAction, replace: bool },
+    Drop {
+        place: Place<'tcx>,
+        target: BasicBlock,
+        unwind: UnwindAction,
+        scope: Option<(DefId, ItemLocalId)>,
```

Comment:
Please also add a comment explaining the meaning of this new field.

Reply:
Added